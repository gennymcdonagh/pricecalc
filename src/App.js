import React, { Component } from 'react';
import './App.css';
import Form from './components/Form/Form';

class App extends Component {

  render() {
    return (
      <div className="App">
        <h2>pricecalc</h2>
        <Form />
      </div>
    );
  }
}

export default App;
