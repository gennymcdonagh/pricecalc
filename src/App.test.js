import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';

it('renders correctly', () => {
  const app = renderer.create(<App />).toJSON();
  expect(app).toMatchSnapshot();
})

it('should calculate the totals correctly', () => {
  const app = shallow(<App />);
  app.instance().calculateTotals({cost:50, items:10,time:10,wage:10,markup:20});

  const correctResult = {
    "Cost to make": 15,
    "Profit": 3,
    "Retail price": 18,
  };
  expect(app.state('resultData')).toMatchObject(correctResult);
})
