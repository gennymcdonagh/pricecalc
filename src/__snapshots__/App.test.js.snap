// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <h2>
    pricecalc
  </h2>
  <div
    className="form"
  >
    <div
      className="number-input"
    >
      <label
        htmlFor="cost"
      >
        Cost of materials
      </label>
      <input
        className="number-input__field"
        id="cost"
        name="cost"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="number"
        value=""
      />
    </div>
    <div
      className="number-input"
    >
      <label
        htmlFor="items"
      >
        Number of items made
      </label>
      <input
        className="number-input__field"
        id="items"
        name="items"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="number"
        value="1"
      />
    </div>
    <div
      className="number-input"
    >
      <label
        htmlFor="time"
      >
        Hours to make an item
      </label>
      <input
        className="number-input__field"
        id="time"
        name="time"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="number"
        value=""
      />
    </div>
    <div
      className="number-input"
    >
      <label
        htmlFor="wage"
      >
        Hourly wage
      </label>
      <input
        className="number-input__field"
        id="wage"
        name="wage"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="number"
        value=""
      />
    </div>
    <div
      className="number-input"
    >
      <label
        htmlFor="markup"
      >
        Retail profit markup %
      </label>
      <input
        className="number-input__field"
        id="markup"
        name="markup"
        onBlur={[Function]}
        onChange={[Function]}
        required={true}
        type="number"
        value=""
      />
    </div>
  </div>
</div>
`;
